src/themes/default/App.vue

```diff
@@ -26,6 +27,14 @@ export default {
       return `${(this.$route.meta.layout || 'default')}-layout`
     }
   },
+  methods: {
+    fetchMenuData () {
+      return this.$store.dispatch('category/list', { includeFields: config.entities.optimize ? config.entities.category.includeFields : null })
```

Comment:
If we’re moving this to layout we should check all asyncDatas as I’m almost sure we’re also fetching categories in the Product.js and Category.js. Next thing is the sequence to make sure that App.vue serverPrefetch is being caked before Product and Category - otherwise it could geberate some side effects / bugs

Reply:
Moreover it should be fetched in Default and maybe Simple layout burn nit in Minkmal layout which is used on error page - which is displayed most often in case that there is no elasircsearch connectivity. We can’t  have this in App.vue but rather in specific layout which is dynamically loaded by App.vue

Reply:
Regarding other pages - you're right I'll check it.
Regarding the sequence - what kind of side effects or bugs you are talking about? is this list from menu used anywhere else? Every page is working correctly

Reply:
I think that layout based prefetch will be always called first so it’s probably always safe. I think it will be safe.